src/components/m-accordion-item/js/accordion-item.js

```diff
@@ -0,0 +1,163 @@
+import UiEvents from '../../../js/ui-events';
+import on from '../../../js/on';
+import { requestAnimationFrame } from '../../../js/request-animation-frame';
+import {
+  add,
+  remove,
+} from '../../../js/class-list';
+import {
+  publish,
+  subscribe,
+} from '../../../js/pubsub';
+
+class AccordionItem extends UiEvents {
+  static DEFAULTS = {
+    container: '.js-accordion-item',
+    toggle: '.js-accordion-item__toggle',
+    body: '.js-accordion-item__body',
+    isOpen: 'is-accordion-item-open',
+  };
+
+  constructor(rootNode, options) {
+    // eslint-disable-next-line no-param-reassign
+    options = {
+      ...AccordionItem.DEFAULTS,
+      ...options,
+    };
+
+    super(rootNode, options);
+
+    this.isOpen = false;
+
+    this.options = options;
+    this.rootNode = rootNode;
+
+    this.handleTransitionEnd = this.handleTransitionEnd.bind(this);
+    this.handleToggleClick = this.handleToggleClick.bind(this);
+    this.toggleState = this.toggleState.bind(this);
+
+    this.init();
+  }
+
+  init() {
+    this.toggle = this.rootNode.querySelector(this.options.toggle);
+    this.body = this.rootNode.querySelector(this.options.body);
+
+    this.on();
+  }
+
+  on() {
+    this.off();
+
+    this.offToggleClicked = on(this.toggle, 'click', this.handleToggleClick);
+  }
+
+  off() {
+    if (this.offToggleClicked) {
+      this.offToggleClicked();
+    }
+  }
+
+  set contextNode(value) {
+    this._contextNode = value;
+
+    this.onContextEnabled();
+  }
+
+  onInteractive() {
+    this.offInteractive();
+
+    this.unTransitionEnd = on(this.body, 'transitionend', this.handleTransitionEnd);
+  }
+
+  offInteractive() {
+    if (this.unTransitionEnd) {
+      this.unTransitionEnd();
+    }
+  }
+
+
+  toggleState({ detail: node }) {
+    if (this.rootNode === node) {
+      if (this.isOpen) {
+        this.close();
+      } else {
+        this.open();
+      }
+    } else if (this.isOpen) {
+      this.close();
+    }
+  }
+
+  open() {
+    const parentNode = this.rootNode;
+    const { lastElementChild } = parentNode;
+
+    this.isOpen = true;
+
+    lastElementChild.style.overflow = 'scroll';
```

Comment:
@TheDadi 
@LucaMele 
I think it makes sense to extract this expand/collapse magic into a global JS util.